File: Core/Classes/PaginationParams.cs
using Core.Interfaces;

namespace Core.Classes
{
    public class PaginationParams : IPaginationParams
    {
        private const int MaxPageSize = 50;
        public int PageIndex { get; set; } = 1;
        private int _pageSize = 10;
        public int PageSize
        {
            get => _pageSize;
            set => _pageSize = value < 50 ? value : MaxPageSize;
        }
        public string Sort { get; set; }
        public string SortToLower()
        {
            return string.IsNullOrWhiteSpace(Sort) ? null : Sort.Trim().ToLower();
        }
    }
}

File: client/src/app/shared/classes/pagination-params.ts
import { IPaginationParams } from '../interfaces/i-pagination-params';

export class PaginationParams implements IPaginationParams {
    pageIndex = 1;
    pageSize = this.itemsPerPageOptions()[1];

    itemsPerPageOptions() {
        return [5, 10, 25, 50, 100];
    }
}
